xarray/core/accessor_str.py

```diff
@@ -865,49 +1061,838 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+                list(['abra', 'ka\\nda\\tbra'])]], dtype=object)
+        Dimensions without coordinates: X, Y
+
+        Split as many times as needed and put the results in a list
+
+        >>> values.str.split(dim=None)
+        <xarray.DataArray (X: 2, Y: 3)>
+        array([[list(['abc', 'def']), list(['spam', 'eggs', 'swallow']),
+                list(['red_blue'])],
+               [list(['test0', 'test1', 'test2', 'test3']), list([]),
+                list(['abra', 'ka', 'da', 'bra'])]], dtype=object)
+        Dimensions without coordinates: X, Y
+
+        Split only on spaces
+
+        >>> values.str.split(dim="ZZ", sep=" ")
+        <xarray.DataArray (X: 2, Y: 3, ZZ: 3)>
+        array([[['abc', 'def', ''],
+                ['spam\\t\\teggs\\tswallow', '', ''],
+                ['red_blue', '', '']],
+        <BLANKLINE>
+               [['test0\\ntest1\\ntest2\\n\\ntest3', '', ''],
+                ['', '', ''],
+                ['abra', '', 'ka\\nda\\tbra']]], dtype='<U24')
+        Dimensions without coordinates: X, Y, ZZ
+
+        See Also
+        --------
+        DataArray.str.rsplit
+        str.split
+        pandas.Series.str.split
+        """
+        if sep is not None:
+            sep = self._obj.dtype.type(sep)
+
+        if dim is None:
+            f = lambda x: x.split(sep, maxsplit)
+            return self._apply(f, dtype=np.object_)
+
+        if not self._obj.size:
+            return self._obj.copy().expand_dims({dim: 0}, -1)
+
+        f_count = lambda x: max(len(x.split(sep, maxsplit)), 1)
+        maxsplit = self._apply(f_count, dtype=np.int_).max().data.tolist() - 1
+
+        def _dosplit(mystr, sep=sep, maxsplit=maxsplit, dtype=self._obj.dtype):
+            res = mystr.split(sep, maxsplit)
+            if len(res) < maxsplit + 1:
+                res += [""] * (maxsplit + 1 - len(res))
+            return np.array(res, dtype=dtype)
+
+        # dtype MUST be object or strings can be truncated
+        # See: https://github.com/numpy/numpy/issues/8352
+        return self._apply(
+            _dosplit,
+            dtype=np.object_,
+            output_core_dims=[[dim]],
+            output_sizes={dim: maxsplit},
+        ).astype(self._obj.dtype.kind)
+
+    def rsplit(
+        self,
+        dim: Optional[Hashable],
+        sep: str = None,
+        maxsplit: int = -1,
+    ) -> Any:
+        """
+        Split strings in a DataArray around the given separator/delimiter `sep`.
+
+        Splits the string in the DataArray from the end,
+        at the specified delimiter string.
+
+        This is equivalent to :meth:`str.rsplit`.
+
+        Parameters
+        ----------
+        dim : Hashable or `None`
+            Name for the dimension to place the results in.
+            If `None`, place the results as list elements in an object DataArray
+        sep : str, default is split on any whitespace.
+            String to split on.
```

Comment:
How does that work if the default is `None`? Could you set this to `"\w"`?

Reply:
It is handled by the `str.split` method.  The unit tests check this.